rust_filter/src/graph_utils.rs

```diff
@@ -92,12 +96,36 @@ pub fn generate_trace_graph_from_headers(
             prev_node_handle = node_str_to_node_handle[node_str];
         }
     }
+
+    // now insert properties:
+    // do we have any?
+    let comma_index = properties_header.find(",");
+    if !comma_index.is_none() {
+        let properties_iterator = properties_header.split(",");
+        for statement in properties_iterator {
+            if !statement.find("==").is_none() {
+                let mut values_iterator = statement.split("==");
+                let property_with_node = values_iterator.next().unwrap().clone().to_string();
+                let mut period_iterator = property_with_node.split(".");
+                let node = period_iterator.next().unwrap();
+                let mut property = String::new();
+                for quality in period_iterator {
+                    property.push_str(&quality);
+                    property.push_str(".");
+                }
+                property.truncate(property_with_node.len()-node.len()-1); // we subtract an extra 2, one for the trailing period on the last iteration of the for loop
```

Comment:
https://doc.rust-lang.org/std/primitive.slice.html#method.split_last
Is probably a good way to handle this.

Reply:
While that is prettier, you can't own a slice.  So you'd wind up reassigning property to a subset of property that it doesn't own, which doesn't make sense, because then the original property is then lost.  Technically all we're doing is getting rid of the trailing period  so I changed it to property.pop() and hopefully that's more readable.